src/accessibility/gridOutput.js

```diff
@@ -0,0 +1,159 @@
+ */
+import p5 from '../core/main';
+
+let ingredients = {};
+let background;
+
+//creates html structure
+p5.prototype._createGridOutput = function(cIdT) {
+  let inner =
+    'Grid Output<p id="' +
+    cIdT +
+    'Summary" aria-label="grid output summary"><table id="' +
+    cIdT +
+    'OD" summary="grid output content"></table><ul id="' +
+    cIdT +
+    'SD" aria-label="grid output shape details"></ul>';
+  return inner;
+};
+
+//creates output content
+p5.prototype._updateGridOutput = function(cnvId, ing, bkgr) {
+  ingredients = ing;
+  background = bkgr;
+  let cIdT = cnvId + 'grOut';
+  let innerSDs = this._buildGridSDs(cIdT);
+  let innerSummary = this._buildGrSummary(innerSDs.numShapes);
+  let innerMap = this._buildGrMap(cIdT);
+  if (innerSummary !== document.getElementById(cIdT + 'Summary').innerHTML) {
+    document.getElementById(cIdT + 'Summary').innerHTML = innerSummary;
+  }
+  if (innerMap !== document.getElementById(cIdT + 'OD').innerHTML) {
+    document.getElementById(cIdT + 'OD').innerHTML = innerMap;
+  }
+  if (innerSDs !== document.getElementById(cIdT + 'SD').innerHTML) {
+    document.getElementById(cIdT + 'SD').innerHTML = innerSDs.details;
+  }
+};
+
+//creates spatial grid
+p5.prototype._buildGrMap = function(cId) {
+  let el = 0;
+  let table = '';
+  let cells = Array.apply(null, Array(10)).map(function() {});
+  for (let r in cells) {
+    cells[r] = Array.apply(null, Array(10)).map(function() {});
+  }
+  for (let x in ingredients) {
+    for (let y in ingredients[x]) {
+      el++;
+      if (
+        cells[ingredients[x][y].loc.locY][ingredients[x][y].loc.locX] ===
+        undefined
+      ) {
+        cells[ingredients[x][y].loc.locY][ingredients[x][y].loc.locX] =
+          '<a href="#' +
+          cId +
+          'shape' +
+          el +
+          '">' +
+          ingredients[x][y].color +
+          ' ' +
+          x +
+          '</a>';
+      } else {
+        cells[ingredients[x][y].loc.locY][ingredients[x][y].loc.locX] =
+          cells[ingredients[x][y].loc.locY][ingredients[x][y].loc.locX] +
+          '  <a href="#' +
+          cId +
+          'shape' +
+          el +
+          '">' +
+          ingredients[x][y].color +
+          ' ' +
+          x +
+          '</a>';
+      }
+    }
+  }
+  for (let _r in cells) {
+    let row = '<tr>';
+    for (let c in cells[_r]) {
+      row = row + '<td>';
+      if (cells[_r][c] !== undefined) {
+        row = row + cells[_r][c];
+      }
+      row = row + '</td>';
+    }
+    row = row + '</tr>';
+    table = table + row;
+  }
+  return table;
+};
+
+//creates grid summary
+p5.prototype._buildGrSummary = function(numShapes) {
+  let text =
+    background +
+    ' canvas, ' +
+    Math.round(this.width) +
+    ' by ' +
+    Math.round(this.height) +
+    ' pixels, contains ' +
+    numShapes[0] +
+    ' shapes: ' +
+    numShapes[1];
+  return text;
+};
+
+//creates list of shapes
+p5.prototype._buildGridSDs = function(cId) {
```

Comment:
similar here, to make this more readable, you could write out `_buildGridShapeDetails`